docs/docs/05-migrations/15-support-stackbuilder.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Support the new StackBuilder primitive for Wiring Middlewares in the chain application
+sidebar_label: Support StackBuilder Wiring
+sidebar_position: 1
+slug: /migrations/support-stackbuilder
+---
+
+# Migration for Chains wishing to use StackBuilder
+
+The StackBuilder struct is a new primitive for wiring middleware in a simpler and less error-prone manner. It is not a breaking change thus the existing method of wiring middleware still works, though it is highly recommended to transition to the new wiring method.
+
+Refer to the [integration guide](../01-ibc/04-middleware/03-integration.md) to understand how to use this new middleware to improve middleware wiring in the chain application setup.
+
+# Migrations for Application Developers
+
+In order to be wired with the new StackBuilder primitive, applications and middlewares must implement new methods as part of their respective interfaces.
+
+IBC Applications must implement a new `SetICS4Wrapper` which will set the `ICS4Wrapper` through which the application will call `SendPacket` and `WriteAcknowledgement`. It is recommended that IBC applications are initialized first with the IBC ChannelKeeper directly, and then modified with a middleware ICS4Wrapper during the stack wiring. 
+```go
+```
+
+```go
+// SetICS4Wrapper sets the ICS4Wrapper. This function may be used after
+// the module's initialization to set the middleware which is above this
+// module in the IBC application stack.
+// The ICS4Wrapper **must** be used for sending packets and writing acknowledgements
+// to ensure that the middleware can intercept and process these calls.
+// Do not use the channel keeper directly to send packets or write acknowledgements
+// as this will bypass the middleware.
+SetICS4Wrapper(wrapper ICS4Wrapper)
+```
+
+Many applications have a stateful keeper that executes the logic for sending packets and writing acknowledgements. In this case, the keeper in the application must be a **pointer** reference so that it can be modified in place after initialization.
+
+The initialization should be modified to no longer take in an addition `ics4Wrapper` as this gets modified later by `SetICS4Wrapper`. The constructor function must also return a **pointer** reference so that it may be modified in-place by the stack builder.
```

Comment:
Strictly speaking, it doesn't need to actually return a pointer, but it is important that it uses a pointer to the keeper.

Reply:
It does I believe, since we are modulating the underlying application/ics4wrapper of the middleware itself during stack.Build()